cluster/src/test/scala/io/snappydata/tools/LeaderLauncherSpec.scala

```diff
@@ -90,7 +87,10 @@ class LeaderLauncherSpec extends WordSpec with Matchers {
 
         {
           // Stop if already any present
-          SnappyContext.stop()
+          if (SnappyContext.globalSparkContext != null &&
+              !SnappyContext.globalSparkContext.isStopped) {
+            SnappyContext.globalSparkContext.stop
+          }
```

Comment:
The new SnappyContext.stop() does only SnappyContext specific cleanup. Maybe it is better to add an argument there like in StreamingContext:

```scala
def stop(stopSparkContext: Boolean) {
  if (stopSparkContext) {
    val sc = globalSparkContext
    if (sc != null && !sc.isStopped) {
      sc.stop()
    }
  } else {
    // the new code
  }
}
```